rfc/service-retrievals.md

```diff
@@ -0,0 +1,38 @@
+# Service Retrievals
+
+There are a number of occasions where internal services within the Storacha Network need to retrieve data from Storage Nodes:
+
+1. The _Upload Service_ retrieves indexes as part of an `space/index/add` invocation, so that it can publish all the hashes to IPNI.
```

Comment:
This refers to the bitswap/trustless-http-gateway IPNI chain pointing to elastic.dag.house and dag.w3s.link right? For warm storage, we should not be publishing this chain.

We can figure out hot storage later but I would say the eventual goal here is that the gateway (or other CDN) itself is the one publishing this chain, and space/index/add should go to them, and then the client should pay for the one time retrieval